server/parsec/components/postgresql/realm.py

```diff
@@ -100,7 +79,18 @@ def _make_q_lock_realm(for_update: bool = False, for_share=False) -> Q:
     FOR {share_or_update}
 )
 SELECT
-    { q_user_role(organization="realm.organization", realm="realm._id", user_id="$user_id") } role,
+    COALESCE(
+        (
+            SELECT realm_user_role.role
+            FROM realm_user_role
+            WHERE
+                realm_user_role.realm = realm._id
+                AND realm_user_role.user_ = { q_user_internal_id(organization="realm.organization", user_id="$user_id") }
+            ORDER BY certified_on DESC
+            LIMIT 1
+        ),
+        NULL
+    ) AS role,
```

Comment:
Maybe a `JOIN` would be more readable than a subquery? Or maybe it's too tricky to write due to `COALESCE`?

Reply:
Maybe... actually I think I skipped this query because I wasn't sure about it. 

`COALESCE` it supposed to returns the first non-`NULL` element, so:
1. if the 1st element (the Select) is `NULL`, check next element
2. the 2nd element is `NULL`, check next element
3. No more elements, return `NULL` 

So this query
```sql
COALESCE(
    (
        SELECT --...
    ),
    NULL
) AS role,
```
is equivalent to:
```sql
COALESCE(
    (
        SELECT --...
    )
) AS role,
```
which is equivalent to:
```sql
    (
        SELECT --...
    ) AS role,
```

Reply:
I've left the subquery but removed the `COALESCE`